smartsim/entity/application.py

```diff
@@ -247,7 +247,7 @@ def attached_files_table(self) -> str:
         :returns: String version of table
         """
         if not self.files:
-            return "No file attached to this application."
+            raise ValueError("No file attached to this application.")
         return str(self.files)
```

Comment:
@drozt and I were talking and thinking at the user level (e.g. Application, LaunchSettings, etc) we might want to use explicit ``isinstance`` checks to make sure that the correct types are passed in.  We probably can't assume users are doing static type checking.  If the ``isinstance`` fails, we would raise a ``TypeError``.  Thoughts?

Reply:
I can do that!